ui/modules/things/thingMessages.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+/* eslint-disable require-jsdoc */
+import * as API from '../api.js';
+import * as Environments from '../environments/environments.js';
+import * as Utils from '../utils.js';
+import messagesHTML from './thingMessages.html';
+import * as Things from './things.js';
+
+const dom = {
+  inputThingMessageSubject: null,
+  inputThingMessageTimeout: null,
+  inputThingMessageTemplate: null,
+  buttonThingMessageSend: null,
+  buttonThingMessageFavorite: null,
+  ulThingMessageTemplates: null,
+  favIconThingMessage: null,
+};
+
+let acePayload;
+let aceResponse;
+
+/**
+ * Initializes components. Should be called after DOMContentLoaded event
+ */
+export async function ready() {
+  Environments.addChangeListener(onEnvironmentChanged);
+
+  Utils.addTab(
+      document.getElementById('tabItemsThing'),
+      document.getElementById('tabContentThing'),
+      'Message to Thing',
+      messagesHTML,
+  );
+
+  Utils.getAllElementsById(dom);
+
+  acePayload = Utils.createAceEditor('acePayloadThingMessage', 'ace/mode/json');
+  aceResponse = Utils.createAceEditor('aceResponseThingMessage', 'ace/mode/json', true);
+
+
+  dom.buttonThingMessageSend.onclick = () => {
+    Utils.assert(dom.inputThingMessageSubject.value, 'Please give a Subject', dom.inputThingMessageSubject);
+    Utils.assert(dom.inputThingMessageTimeout.value, 'Please give a timeout', dom.inputThingMessageTimeout);
+    dom.buttonThingMessageSend.classList.add('busy');
+    dom.buttonThingMessageSend.disabled = true;
+    messageThing();
+  };
+
+  dom.buttonThingMessageFavorite.onclick = () => {
+    const templateName = dom.inputThingMessageTemplate.value;
+    Utils.assert(templateName, 'Please give a name for the template', dom.inputThingMessageTemplate);
+    Environments.current().messageTemplates['/'] = Environments.current().messageTemplates['/'] || {};
+    if (Object.keys(Environments.current().messageTemplates['/']).includes(templateName) &&
+        dom.favIconThingMessage.classList.contains('bi-star-fill')) {
+      dom.favIconThingMessage.classList.replace('bi-star-fill', 'bi-star');
+      delete Environments.current().messageTemplates['/'][templateName];
+    } else {
+      dom.favIconThingMessage.classList.replace('bi-star', 'bi-star-fill');
+      Environments.current().messageTemplates['/'][templateName] = {
+        subject: dom.inputThingMessageSubject.value,
+        timeout: dom.inputThingMessageTimeout.value,
+        payload: JSON.parse(acePayload.getValue()),
```

Comment:
Storing a template does not work, if the payload is empty (same is true for the messages on features). I did a commit to fix that but I have no permissions to push. Would it work with a patch or how is the procedure here?

Reply:
@thfries yes, a patch would be great 👍 - I unfortunately cannot selectively give you permissions to make changes in this repo.

Reply:
let's try: 
[0001-UI-improvements-review.patch](https://github.com/eclipse-ditto/ditto/files/12910260/0001-UI-improvements-review.patch)